### DISCUSSION TITLE
CSS spliting depending on media queries

### DISCUSSION THREAD

**Discussion by juanchristensen** [vercel/next.js#13324] [Ideas]
Hi everyone,

Resizing the browser window is a relatively rare event from the user perspective ( most of the time we stay with the same window size).

Given that, we could remove the styles with unmatched media queries while doing SSR.

Another, maybe simpler approach, would be to keep only the base styles and media queries on our components, and conditionally import the specific imports when a media query is matched (using conditional dynamic imports).

Thoughts?

**Comment by Sheraff** [comment]
If I may link to an identical idea with a more concrete implementation proposal for future reference: https://github.com/vercel/next.js/discussions/29331
